ephios/user_management/views.py

```diff
@@ -21,47 +30,75 @@ class UserProfileListView(PermissionRequiredMixin, ListView):
     permission_required = "user_management.view_userprofile"
 
 
-class UserProfileCreateView(PermissionRequiredMixin, CreateView):
+class UserProfileCreateView(PermissionRequiredMixin, TemplateView):
     template_name = "user_management/userprofile_form.html"
     permission_required = "user_management.add_userprofile"
     model = UserProfile
-    form_class = UserProfileForm
 
-    def get_success_url(self):
-        messages.success(self.request, _("User added successfully."))
-        return reverse("user_management:userprofile_list")
-
-    def form_valid(self, form):
-        response = super().form_valid(form)
-        userprofile = self.object
-        if userprofile.is_active:
-            mail.send_account_creation_info(userprofile)
-        return response
-
-
-class UserProfileUpdateView(PermissionRequiredMixin, UpdateView):
+    def get_context_data(self, **kwargs):
+        kwargs.setdefault("userprofile_form", UserProfileForm(self.request.POST or None))
+        kwargs.setdefault(
+            "qualification_formset", QualificationGrantFormset(self.request.POST or None)
+        )
+        return super().get_context_data(**kwargs)
+
+    def post(self, request, *args, **kwargs):
+        userprofile_form = UserProfileForm(self.request.POST)
+        qualification_formset = QualificationGrantFormset(self.request.POST)
+        if userprofile_form.is_valid() and qualification_formset.is_valid():
```

Comment:
I'm not sure we'd get all validation-errors with the `and` short-circuiting. Could do: 
```suggestion
        if all((userprofile_form.is_valid(), qualification_formset.is_valid())):
```